oaknational/Oak-Web-Application
# Feat/error reporting

## Description

- adds bugsnag js and react libs to package.json
- creates `ErrorBoundary` component (which is a wrapper round busnag's error boundary)
- creates `createErrorHandler` function, which ideally we'll want to call every time we catch an error, and should be usable both client and server side

## Issue(s)

Fixes #30

## How to test

1. Go to {cloud link}
2. Make the page throw an error (eg add `onclick="throw new Error('manual test')"` to an html element in the devtools
3. Head [over to bugsnag](https://app.bugsnag.com/oak-national-academy/samara)
4. You should see your error coming though at 
5. Locally run the app, and throw an error in a component,
6. You should see the error page show like the screenshot below (you'll have to close the react error modal first)

## Screenshots

How it should now look:

![image](https://user-images.githubusercontent.com/12934669/163195129-05bf1fa7-106a-4e0e-9477-7017b5de737e.png)



## Checklist
